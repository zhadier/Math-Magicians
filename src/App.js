/** @format */
import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Calculator from './components/Calculator';
import Quote from './components/Quote';
import Home from './components/Home';
import Header from './components/Header';
import './App.scss';

const App = () => (
  <>
    <Header />
    <Routes>
      <Route path="math-magicians" element={<Home />} />
      <Route path="math-magicians/calculator" element={<Calculator />} />
      <Route path="math-magicians/quote" element={<Quote />} />
    </Routes>
  </>
);

export default App;
