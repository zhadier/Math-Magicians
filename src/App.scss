@import url('https://fonts.googleapis.com/css2?family=Cinzel:wght@600&family=Roboto&display=swap');
$dark-blue: rgb(19, 19, 77);

*,
*::before,
*::after {
  box-sizing: border-box;
  scroll-behavior: smooth;
  padding: 0%;
  margin: 0%;
}

body {
  background: url(./images/bg.png) center;
}

section {
  display: flex;
  width: 95%;
  margin: 80px auto;
  border: 3px solid;
  border-radius: 10px;
  background: white;
  padding: 30px;
  min-height: 60vh;
}

h1,
h2,
a {
  font-family: 'Cinzel', serif;
  color: $dark-blue;
}

p {
  font-family: 'Roboto', sans-serif;
}
