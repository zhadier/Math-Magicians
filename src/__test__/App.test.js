import React from 'react';
import renderer from 'react-test-renderer';
import {
  cleanup,
  render,
  fireEvent,
  screen,
} from '@testing-library/react';
import App from '../App';

afterEach(() => cleanup());

it('App Renders Correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Clicking on Home link renders home', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Home'));
  expect(screen.getByText('Welcome to our page!')).toBeInTheDocument();
});

it('Clicking on Calculator link renders Calculator', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Calculator'));
  expect(screen.getByText('Let\'s do some math!')).toBeInTheDocument();
});

it('Clicking on Calculator link renders Calculator', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Quote'));
  expect(screen.getByText('Loading...')).toBeInTheDocument();
});

it('Clicking on Calculator link does not render Home', () => {
  render(<App />);
  fireEvent.click(screen.getByText('Calculator'));
  expect(screen.queryByText('Welcome to our page!')).toBeNull();
});
