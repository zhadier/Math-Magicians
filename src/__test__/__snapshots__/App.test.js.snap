// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<header>
  <h1
    className="logo"
  >
    Math Magicians
  </h1>
  <nav
    className="header__nav"
  >
    <ul>
      <li>
        <a
          href="/math-magicians"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/math-magicians/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/math-magicians/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
